src/internal_events/http_scrape.rs

```diff
@@ -0,0 +1,90 @@
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+use super::prelude::{error_stage, error_type, http_error_code};
+
+#[derive(Debug)]
+pub struct HttpScrapeEventsReceived {
+    pub byte_size: usize,
+    pub count: usize,
+    pub url: String,
```

Comment:
Could we [`as_str()`](https://doc.rust-lang.org/std/string/struct.String.html#method.as_str) for more optimization?

Reply:
Along the same lines as https://github.com/vectordotdev/vector/pull/13793#discussion_r939189461
, The `counter!` macros require a static lifetime for the string.

Reply:
And since we can't know that at runtime, I'm not seeing an obvious way to optimize this further.

Reply:
Maybe someone with more exp has an idea otherwise 👍

Reply:
Right, the tags to the metrics are effectively (if not literally) `Cow<'static, str>` meaning either an owned string or a static reference. There's not much better that can be done here. Once I have the registered event framework in place, we can generate handles for any metric that is used repeatedly with the same tags, but I don't think that's the case here.

Reply:
👍 I think that is requirement of the metrics macros that they own they labels.